src/api.ts

```diff
@@ -822,6 +960,46 @@ function createApi({
         )
       }
 
+      let spriteId: string | undefined
+
+      if (style.sprite) {
+        let upstreamSprites:
+          | Awaited<ReturnType<typeof fetchUpstreamSprites>>['sprites']
+          | undefined
+        try {
+          const result = await fetchUpstreamSprites({
+            accessToken,
+            upstreamSpriteUrl: style.sprite,
+          })
+
+          if (
+            [...result.sprites.values()].every((info) => info instanceof Error)
+          ) {
+            throw new FailedUpstreamFetchError(style.sprite)
+          }
+
+          spriteId = result.id
+          upstreamSprites = result.sprites
+        } catch (err) {
+          // This will either be an error about the access token or the inability to fetch any sprites upstream
+          throw err
+        }
+
+        for (const [pixelDensity, spriteInfo] of upstreamSprites.entries()) {
+          // TODO: Should we report the error here?
+          if (spriteInfo instanceof Error) continue
```

Comment:
highlighting this since not entirely sure what to do in this case (i.e. only one of the the upstream requests failed)